api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -425,85 +431,177 @@ public PatientIdentifierType getPatientIdentifierType(Integer patientIdentifierT
+				innerFields.add("p1." + s);
+			} else if (personFieldNames.contains(s)) {
+				if (!outerSelect.contains("person")) {
+					outerSelect += "inner join person t2 on t1.patient_id = t2.person_id ";
+					innerSelect += "inner join person person1 on p1.patient_id = person1.person_id ";
+				}
+
+				AbstractEntityPersister aep = ((AbstractEntityPersister) sessionFactory.getClassMetadata(Person.class));
+				if (aep != null) {
+					String[] properties = aep.getPropertyColumnNames(s);
+					if (properties != null && properties.length >= 1)
+						s = properties[0];
+				}
+
+				whereConditions.add(" t2." + s + " = t5." + s);
+				innerFields.add("person1." + s);
+			} else if (personNameFieldNames.contains(s)) {
+				if (!outerSelect.contains("person_name")) {
+					outerSelect += "inner join person_name t3 on t2.person_id = t3.person_id ";
+					innerSelect += "inner join person_name pn1 on person1.person_id = pn1.person_id ";
+				}
+
+				//Since we are firing a native query get the actual table column name from the field name of the entity
+				AbstractEntityPersister aep = ((AbstractEntityPersister) sessionFactory
+						.getClassMetadata(PersonName.class));
+				if (aep != null) {
+					String[] properties = aep.getPropertyColumnNames(s);
+
+					if (properties != null && properties.length >= 1)
+						s = properties[0];
+				}
+
+				whereConditions.add(" t3." + s + " = t5." + s);
+				innerFields.add("pn1." + s);
+			} else if (identifierFieldNames.contains(s)) {
+				if (!outerSelect.contains("patient_identifier")) {
+					outerSelect += "inner join patient_identifier t4 on t1.patient_id = t4.patient_id ";
+					innerSelect += "inner join patient_identifier pi1 on p1.patient_id = pi1.patient_id ";
+				}
+
+				AbstractEntityPersister aep = ((AbstractEntityPersister) sessionFactory
+						.getClassMetadata(PatientIdentifier.class));
+				if (aep != null) {
+
+					String[] properties = aep.getPropertyColumnNames(s);
+					if (properties != null && properties.length >= 1)
+						s = properties[0];
+				}
+
+				whereConditions.add(" t4." + s + " = t5." + s);
+				innerFields.add("pi1." + s);
+			} else {
+				log.warn("Unidentified attribute: " + s);
 			}
-			
-			if (!attributes.contains("includeVoided")) {
-				where += "and p1.voided = false and p2.voided = false ";
+		}
+
+		String innerFieldsJoined = StringUtils.join(innerFields, ", ");
+		String whereFieldsJoined = StringUtils.join(whereConditions, " and ");
+		String innerWhereCondition = "";
+		if (!attributes.contains("includeVoided")) {
+			innerWhereCondition = " where p1.voided = false ";
+		}
+		String innerQuery = "(Select " + innerFieldsJoined + innerSelect + innerWhereCondition + " group by "
+				+ innerFieldsJoined + " having count(*) > 1" + " order by " + innerFieldsJoined + ") t5";
+		return outerSelect + ", " + innerQuery + " where " + whereFieldsJoined + ";";
+	}
+
+	private void sortDuplicatePatients(List<Patient> patients, List<Integer> patientIds) {
```

Comment:
@dkayiwa : Please refer the comment below to understand what happens if sorting isn't done.